settings.c

```diff
@@ -244,6 +249,26 @@ uint8_t settings_store_global_setting(int parameter, float value) {
     case 40: settings.mag_gap_limit = value; break;
     case 41: settings.mag_gap_enabled = value; break;
     case 42: settings.use_load_cell = value; break;
+    case 46: case 47: case 48: case 49:
+      if (value > TWO_SIXTY_FOURTH) {
+        return(STATUS_INVALID_STATEMENT);
+      }
+      switch (parameter) {
+        case 46:
+          settings.x_microsteps = value;
+          break;
+        case 47:
+          settings.x_microsteps = value;
+          break;
+        case 48:
+          settings.x_microsteps = value;
+          break;
+        case 49:
+          settings.x_microsteps = value;
+          break;
+      }
+      motor_drv_init();
```

Comment:
What would happen if you didn't have SPI enabled?  Seems to me that this doesn't handle both the 3.2 and 4.0 hardware?